aten/src/ATen/native/cuda/Sort.cpp

```diff
@@ -65,6 +65,15 @@ void sort_cuda_kernel(
   const auto self_dtype = self.dtype();
   TORCH_CHECK(self_dtype != ScalarType::ComplexFloat && self_dtype != ScalarType::ComplexDouble,
     "Sort currently does not support complex dtypes on CUDA.");
+#if defined(USE_ROCM)
+  // ROCm has undefined behavior for non-standard bools. Here we are converting bool to uint8 which will
+  // convert false to 0 and true or any non-zero value to a 1. copy_ on const Tensors only changes the
+  // data in the tensor and not the metadata.
+  // That's why, tensor's dtype stays as bool. It just becomes a standard bool.
+  if (self_dtype == ScalarType::Bool) {
```

Comment:
This will make the code using regular bools slower, wouldn't it?

Reply:
yeah, but probably better than incorrect results :)